lib/rules/attributes-order.js

```diff
@@ -0,0 +1,506 @@
+import { builders as b } from 'ember-template-recast';
+
+import createErrorMessage from '../helpers/create-error-message.js';
+import isValidConfigObjectFormat from '../helpers/is-valid-config-object.js';
+import replaceNode from '../helpers/replace-node.js';
+import Rule from './_base.js';
+
+const ERROR_MESSAGE = 'Your config does not match the allowed values';
+
+const AttributeType = {
+  ARGUMENTS: 'arguments',
+  ATTRIBUTES: 'attributes',
+  MODIFIERS: 'modifiers',
+  SPLATTRIBUTES: 'splattributes',
+};
+
+const DEFAULT_CONFIG = {
+  alphabetize: true,
+  attributeOrder: [AttributeType.ARGUMENTS, AttributeType.ATTRIBUTES, AttributeType.MODIFIERS],
+};
+
+export function createAttributesOrderErrorMessage(config) {
+  return createErrorMessage('attributes-order', ERROR_MESSAGE, config);
+}
+
+export default class AttributesOrder extends Rule {
+  appliedOrder = undefined;
+
+  parseConfig(config) {
+    const errorMessage = createAttributesOrderErrorMessage(config);
+    if (!isValidConfigObjectFormat(config, DEFAULT_CONFIG)) {
+      throw new Error(errorMessage);
+    }
+
+    if (config === false) {
+      return false;
+    }
+
+    if (typeof config === 'object') {
+      return { ...DEFAULT_CONFIG, ...config };
+    }
+
+    return DEFAULT_CONFIG;
+  }
+
+  areSplattributesSurrounded(nodeTokens) {
+    const splattributeLastIndex = nodeTokens[AttributeType.SPLATTRIBUTES].last_index;
+    if (splattributeLastIndex < 0) {
+      return false;
+    }
+    const maxIndex = Math.max(
+      ...Object.values(nodeTokens).map(
+        (nodeToken) => nodeToken.last_index || Number.NEGATIVE_INFINITY
+      )
+    );
+    const minIndex = Math.min(
+      ...Object.values(nodeTokens).map(
+        (nodeToken) => nodeToken.first_index || Number.POSITIVE_INFINITY
+      )
+    );
+    return splattributeLastIndex > minIndex && splattributeLastIndex < maxIndex;
+  }
+
+  /**
+   *
+   * @param {object} nodeTokens
+   * @returns {array} The order of attributes given a particular node
+   * If splattributes are in the first or last position they should remain in that order
+   */
+  getAppliedOrder(nodeTokens) {
+    if (this.appliedOrder) {
+      return this.appliedOrder;
+    }
+    const maxLastIndex = Math.max(
+      ...Object.values(nodeTokens)
+        .map((nodeToken) => nodeToken.last_index)
+        .filter((i) => Number.isInteger(i))
+    );
+    const minLastIndex = Math.min(
+      ...Object.values(nodeTokens)
+        .map((nodeToken) => nodeToken.last_index)
+        .filter((i) => Number.isInteger(i))
+    );
+
+    const attributeOrderMinusSplattributes = [
+      ...this.config.attributeOrder.filter(
+        (attribute) => attribute !== AttributeType.SPLATTRIBUTES
+      ),
+    ];
+
+    const splattributeLastIndex = nodeTokens[AttributeType.SPLATTRIBUTES]?.last_index;
```

Comment:
CI is failing because `?.` isn't available yet since we still support Node 12. Can you fix that?